src/Uno.UI/UI/Xaml/DependencyObjectExtensions.cs

```diff
@@ -95,12 +91,25 @@ internal static Style StoreGetImplicitStyle(this IDependencyObjectStoreProvider
 
 		internal static IEnumerable<object> GetParents(this object dependencyObject)
 		{
-			var parent = dependencyObject.GetParent();
-			while (parent != null)
+			object current = dependencyObject;
+			do
 			{
-				yield return parent;
-				parent = parent.GetParent();
-			}
+				var parent = current.GetParent();
+				if (parent == null)
+				{
+					// Try to use VisualTreeHelper logic as fallback
+					// this accounts for Selector items within ItemsControl
+					parent = VisualTreeHelper.GetParent(current as DependencyObject);
+				}
+
+				if (parent is not null)
+				{
+					yield return parent;
+				}
+
+				current = parent;
+
+			} while (current != null);
```

Comment:
@jeromelaban @dr1rrb @carldebilly Not sure about this change - should `GetParents()` only include "real" parents and stop when `GetParent()` returns `null` or include the `ItemsControl` as parent for `SelectorItem`?

Reply:
This method is an internal version which only tracks DependencyObject hierarchy, not visual tree ones. If VisualTreeHelper needs to behave different, the specificity should be in VisualTreeHelper.

Reply:
Undone